### ISSUE TITLE
Add thresholding algorithms submodule

### ISSUE THREAD

**Issue by jmetz** [rust-cv/ndarray-vision#13]
Ideas for algorithms to add could come from Matlab/Scikit-image/ImageJ threshold algorithms. 

To start we could add some a couple of standard ones - I've been working on mean and Otsu's method implementations.

**Comment by xd009642** [comment]
@jmetz are there any other thresholding algorithms you'd like in particular?

**Comment by jmetz** [comment]
I use Scikit Image a lot - and often use e.g Li threshold algorithm (https://github.com/scikit-image/scikit-image/). In principle though I could imagine implementing all of the standard threshold algorithms as a long term goal. 

ImageJ also contains a good selection of threshold algorithms to pick from: https://imagej.net/Auto_Threshold

**Comment by xd009642** [comment]
Yeah I've got no issue with implementing them all it's just a matter of time (or getting the PRs :smile: )
